tests/xgboost/test_xgboost_model_export.py

```diff
@@ -452,3 +480,49 @@ def test_pyfunc_serve_and_score_sklearn(model):
     )
     scores = pd.read_json(resp.content, orient="records").values.squeeze()
     np.testing.assert_array_equal(scores, model.predict(X.head(3)))
+
+
+@pytest.mark.large
+def test_load_pyfunc_succeeds_for_older_models_with_pyfunc_data_field(xgb_model, model_path):
+    """
+    This test verifies that xgboost models saved in older versions of MLflow are loaded
+    successfully by ``mlflow.pyfunc.load_model``. These older models specify a pyfunc ``data``
+    field referring directly to a serialized scikit-learn model file. In contrast, newer models
+    omit the ``data`` field.
+    """
+    model = xgb_model.model
+    mlflow.xgboost.save_model(xgb_model=model, path=model_path)
+
+    model_conf_path = os.path.join(model_path, "MLmodel")
+    model_conf = Model.load(model_conf_path)
+    pyfunc_conf = model_conf.flavors.get(pyfunc.FLAVOR_NAME)
+    xgboost_conf = model_conf.flavors.get(mlflow.xgboost.FLAVOR_NAME)
+    assert xgboost_conf is not None
+    assert "model_class" in xgboost_conf
+    assert "data" not in xgboost_conf
+    assert pyfunc_conf is not None
+    assert "model_class" in pyfunc_conf
+    assert pyfunc.DATA not in pyfunc_conf
+
+    pyfunc.add_to_model(model_conf, loader_module="mlflow.xgboost", data="model.xgb")
+    model_conf.add_flavor(mlflow.xgboost.FLAVOR_NAME, data="model.xgb")
+    model_conf.save(model_conf_path)
+    model_conf = Model.load(model_conf_path)
+    xgboost_conf = model_conf.flavors.get(mlflow.xgboost.FLAVOR_NAME)
+    assert "data" in xgboost_conf
+    assert xgboost_conf["data"] == "model.xgb"
```

Comment:
I have to add `data` to MLmodel and save it again to test if `_load_model` can successfully load previously logged XGBoost Booster models. I am not sure if [L465](https://github.com/mlflow/mlflow/commit/aa563fbdb8dc35cc99cb4e73e7e057a16b76acff#diff-5e72f29b7d4bc3e2962472dea49037d8585cd85b1cc3f0dc3b23bce0c27b0459R465) really tests backward compatibility, since `conf` is reloaded inside `pyfunc.load_model()`, and updating `pyfunc_conf` outside doesn't change `conf`.